test/evp_kdf_test.c

```diff
@@ -70,24 +70,55 @@ static int test_kdf_pbkdf2(void)
 {
     int ret;
     EVP_KDF_CTX *kctx;
-    unsigned char out[32];
+    unsigned char out[25];
```

Comment:
Would this be better declared later using `sizeof(expected)` to avoid magic numbers?

Reply:
I dont think this makes much difference. the expected[] still needs to be a fixed size. The KAT that it is taken from has a fixed length, so the current way ensures the outlength is correct.. even if the expected array was incorrect and got zero padded. Either way the KAT would fail and the issue would be resolved.